src/polkadot/PolkadotTools.ts

```diff
@@ -92,11 +103,27 @@ export class PolkadotTools implements EdgeCurrencyTools {
 
   async derivePublicKey(walletInfo: EdgeWalletInfo): Promise<JsonObject> {
     const { pluginId } = this.currencyInfo
+    await cryptoWaitReady()
     const keyring = new Keyring({ ss58Format: this.networkInfo.ss58Format })
-    const pair = keyring.addFromUri(walletInfo.keys[`${pluginId}Mnemonic`])
-    return {
-      publicKey: pair.address
+
+    const mnemonic = walletInfo.keys[`${pluginId}Mnemonic`]
+    const privateKeyHex = walletInfo.keys[`${pluginId}Key`]
+
+    if (typeof mnemonic === 'string') {
+      const pair = keyring.addFromUri(mnemonic)
+      return { publicKey: pair.address }
+    }
+
+    if (typeof privateKeyHex === 'string') {
+      const hex = privateKeyHex.replace(/^0x/i, '')
+      const bytes = Buffer.from(hex, 'hex')
```

Comment:
There are still instances of using Buffer across the code but the preferred way to handle this data is to use base16.parse and base16.stringify